lib/modules/manager/sveltos/extract.ts

```diff
@@ -0,0 +1,96 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
+import { trimTrailingSlash } from '../../../util/url';
+import { parseYaml } from '../../../util/yaml';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import { isOCIRegistry, removeOCIPrefix } from '../helmv3/oci';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { getDep } from '../dockerfile/extract';
+import { ProfileDefinition, type SveltosHelmSource } from './schema';
+import { removeRepositoryName } from './util';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  config?: ExtractConfig,
+): PackageFileContent | null {
+  let definitions: ProfileDefinition[];
+  try {
+    definitions = parseYaml(content, {
+      customSchema: ProfileDefinition,
+    });
+  } catch (err) {
+    logger.debug({ err, packageFile }, 'Failed to parse Sveltos definition.');
+    return null;
+  }
+
+  const deps = definitions.flatMap((definition) => {
+    // Use zod's safeParse method to check if the object matches the ProfileDefinition schema
+    const result = ProfileDefinition.safeParse(definition);
+    if (result.success) {
+      return processAppSpec(result.data, config);
+    }
+    return [];
```

Comment:
extract util function

Reply:
refactored the function (i hope that was you were talking about)